shapeshift/web
# fix: mixpanel use USD amounts instead of user-currency amounts

## Description

This PR:

- brings in new USD selector in addition to the current user-currency ones for donation and sell amounts, both in old and new swapper
- sends the data to mixpanel in USD both for donations and sell amounts *in addition to* the data in user-currency
- sends the `selectedCurrency` in

## Pull Request Type


## Issue (if applicable)



closes #4866

## Risk



If we got this wrong, tracking will be unreliable - test accordingly

## Testing



- Make a few trades, both with and without donations, both with USD and another currency
- Ensure the fiat amounts (sell asset/donation) are correctly passed in in USD and selected currency, alongside the selected currency

### Engineering



- ☝🏽 

### Operations



- ☝🏽 

## Screenshots (if applicable)


### CoW - no donation

<img width="613" alt="Screenshot 2023-07-05 at 17 38 06" src="https://github.com/shapeshift/web/assets/17035424/b3436075-6858-4291-9e15-749c2d9ca6f6">
<img width="378" alt="image" src="https://github.com/shapeshift/web/assets/17035424/ecf61407-91f7-4b59-a0e1-9ed96bfd0bf7">
<img width="409" alt="image" src="https://github.com/shapeshift/web/assets/17035424/f650353f-beb0-4bf6-bb33-31d926257ad9">
<img width="352" alt="image" src="https://github.com/shapeshift/web/assets/17035424/3d826acc-358e-4a33-9eaf-d46bfe3221b4">

### 1inch - with donation

<img width="640" alt="image" src="https://github.com/shapeshift/web/assets/17035424/1782bbc8-a3e4-41b7-b705-bab85f848899">
<img width="366" alt="image" src="https://github.com/shapeshift/web/assets/17035424/d6a77af9-8eb1-40cc-bed9-6c81ab328e89">
<img width="407" alt="image" src="https://github.com/shapeshift/web/assets/17035424/54765a6f-eb4f-462c-b8b1-0ccc24e858fd">
<img width="396" alt="image" src="https://github.com/shapeshift/web/assets/17035424/4f4fac67-8992-4943-b491-5ea71d68168a">
